src/common/collection.c

```diff
@@ -470,57 +555,115 @@ gboolean dt_collection_get_sort_descending(const dt_collection_t *collection)
+
+  switch(collection->params.sortSecondOrder)/*build ORDER BY string for second order*/
+  {
+        case DT_COLLECTION_SORT_DATETIME:
+          second_order = dt_util_dstrcat(NULL, "datetime_taken %s", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_RATING:
+          second_order = dt_util_dstrcat(NULL, "flags & 7 %s", (collection->params.descending ? "" : "DESC"));
+          break;
+
+        case DT_COLLECTION_SORT_FILENAME:
+          second_order = dt_util_dstrcat(NULL, "filename %s", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_ID:
+          second_order = dt_util_dstrcat(NULL, "mi.id %s", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_COLOR:
+          second_order = dt_util_dstrcat(NULL, "color %s", (collection->params.descending ? "" : "DESC"));
+          break;
+
+        case DT_COLLECTION_SORT_GROUP:
+          second_order = dt_util_dstrcat(NULL, "group_id %s, mi.id-group_id != 0", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_PATH:
+          second_order = dt_util_dstrcat(NULL, "folder %s, filename %s", (collection->params.descending ? "DESC" : ""), (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_CUSTOM_ORDER:
+          second_order = dt_util_dstrcat(NULL, "position %s", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_TITLE:
+        case DT_COLLECTION_SORT_DESCRIPTION:/*same sorting for TITLE and DESCRIPTION -> Fall through*/
+          second_order = dt_util_dstrcat(NULL, "m.value %s, caption %s", (collection->params.descending ? "DESC" : ""), (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_ASPECT_RATIO:
+          second_order = dt_util_dstrcat(NULL, "aspect_ratio %s", (collection->params.descending ? "DESC" : ""));
+          break;
+
+        case DT_COLLECTION_SORT_SHUFFLE:
+          /* do not remember shuffle for second order */
+          if( !second_order ) second_order = dt_util_dstrcat(NULL, "filename %s", (collection->params.descending ? "DESC" : ""));/*only set if not yet initialized*/
+          break;
+
+        case DT_COLLECTION_SORT_NONE:
+          // shouldn't happen
+          second_order = dt_util_dstrcat(NULL, "filename %s", (collection->params.descending ? "DESC" : ""));
+          break;
+  }
+
 
   if(collection->params.descending)
   {
     switch(collection->params.sort)
     {
       case DT_COLLECTION_SORT_DATETIME:
-        sq = dt_util_dstrcat(sq, ORDER_BY_QUERY, "datetime_taken DESC, filename DESC, version DESC");
+        sq = dt_util_dstrcat(sq, "%s datetime_taken DESC, %s, filename DESC, version DESC", ORDER_BY_QUERY, second_order);
```

Comment:
Better write "ORDER BY datatime_taken DESC, %s...", second_order.

Here and all other occurrences.

Reply:
done.